src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationKeyComparer.cs

```diff
@@ -41,8 +42,8 @@ public int Compare(string x, string y)
                 int value1 = 0;
                 int value2 = 0;
 
-                bool xIsInt = x != null && int.TryParse(x, out value1);
-                bool yIsInt = y != null && int.TryParse(y, out value2);
+                bool xIsInt = x != null && int.TryParse(x, NumberStyles.None, CultureInfo.InvariantCulture, out value1);
+                bool yIsInt = y != null && int.TryParse(y, NumberStyles.None, CultureInfo.InvariantCulture, out value2);
```

Comment:
Why is this safe?  If the inputs here are coming from developer-written config files, couldn't those contain culture-specific things, like negative signs?  And NumberStyles.None means that whitespace will no longer be ignored at the start/end... has that somewhere already been handled?

Reply:
Maybe I'm not aware of some scenarios, but I was thinking about this in the context of Configuration.Bind when I wrote it:
Only positive integers are usable as array indices, everything else is just a string. If leading and trailing whitespace are ignored for strings as well, then this would break some scenarios, yes.